src/main/java/uk/nhs/ctp/security/SupplierTokenResolver.java

```diff
@@ -0,0 +1,68 @@
+package uk.nhs.ctp.security;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+import uk.nhs.cactus.common.security.CactusToken;
+import uk.nhs.ctp.entities.CdssSupplier;
+import uk.nhs.ctp.entities.EmsSupplier;
+import uk.nhs.ctp.service.CdssSupplierService;
+import uk.nhs.ctp.service.EmsSupplierService;
+
+@Component
+@RequiredArgsConstructor
+public class SupplierTokenResolver {
+
+  @Value("${fhir.server}")
+  private String fhirServer;
+
+  @Value("${blob.server}")
+  private String blobServer;
+
+  @Value("${ems.fhir.server}")
+  private String emsFhirServer;
+
+  @Value("${cactus.cdss}")
+  private String cactusCdss;
```

Comment:
I don't think this is a single baseURL now? We will need to update the registration result to account for multiple endpoints.

For this situation I would leave the CDSS out altogether and rely on providing the token, which would then allow us to test that the interceptor is working.

Reply:
Good point. We'll also need to add the 2.0 CDSS to the result of supplier registration at some point.